src/main/java/com/epam/test_generator/entities/api/ProjectTrait.java

```diff
@@ -0,0 +1,27 @@
+package com.epam.test_generator.entities.api;
+
+import com.epam.test_generator.services.exceptions.ProjectClosedException;
+
+public interface ProjectTrait {
+
+    void setActive(boolean active);
+
+    boolean isActive();
+
+    Long getId();
+
+    default void checkIsActive() {
```

Comment:
What if I need to know if the Project is active? In that case I should call this method and catch the exception thrown?

Reply:
Oh, I see that we also have `isActive()`. Then let's move that logic into service